File: clients/javascript/src/models/msgPublishOutMsg.ts
// this file is @generated





export interface MsgPublishOutMsg {
    topic: string;
offset: number;
}

export const MsgPublishOutMsgSerializer = {
    _fromJsonObject(object: any): MsgPublishOutMsg {
        return {
            topic: object['topic'],
            offset: object['offset'],
            };
    },

    _toJsonObject(self: MsgPublishOutMsg): any {
        return {
            'topic': self.topic,
            'offset': self.offset,
            };
    }
}

File: clients/rust/src/models/msg_publish_out_msg.rs
// this file is @generated
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, Deserialize, Serialize)]
pub struct MsgPublishOutMsg {
    pub topic: String,

    pub offset: u64,
}

impl MsgPublishOutMsg {
    pub fn new(topic: String, offset: u64) -> Self {
        Self { topic, offset }
    }
}
